### ISSUE TITLE
template matching example

### ISSUE THREAD

**Issue by developer239** [justadudewhohacks/opencv4nodejs#135]
Hey,
this looks really cool. Does template matching work? :)

**Comment by justadudewhohacks** [comment]
Hey,
yes matchTemplate is implemented: https://opencv4nodejs.herokuapp.com/docs/core#Mat-matchTemplate.

**Comment by developer239** [comment]
This looks awesome. I am gonna definitely try it out this weekend and maybe create a pull request with working javascript example.

And is performance improved? I have a project with template matching and haar object detection. I use the haar detection on web camera. Is it possible to choose whether to use cpu/gpu when using template matching/haar cascade detection? :)

**Comment by justadudewhohacks** [comment]
What do you mean by performance improved? You can expect the same performance as with the C++ or Python library.

So far I have only added a ocl GPU version of detectMultiScale. In general if you want to more GPU bindings you can add them or I can add them for you. 

However I will only add the OpenCV GPU functions working with OpenCL as I do not have access to a NVIDIA GPU in order to implement functions from the CUDA API. If you do, you can of course add CUDA bindings if you want.

**Comment by developer239** [comment]
I just created #138 this library looks awesome. :)

I feel like template matching has changed a lot. I was always using `"opencv": "5.0.0",` because `matchTemplate` method in that version returns an array of coordinates, not the black and white matrice.

When finding waldo I had to use method 5. I guess it is because I lack some information about how the template matching actually works in opencv.
